packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx

```diff
@@ -253,216 +200,248 @@ export class MonitorATMServicesViewImpl extends React.PureComponent<TProps, Stat
       fetchAllServiceMetrics(currentService, metricQueryPayload);
       fetchAggregatedServiceMetrics(currentService, metricQueryPayload);
 
-      this.setState({ serviceOpsMetrics: undefined, searchOps: '' });
+      setServiceOpsMetrics(undefined);
+      setSearchOps('');
     }
-  }
-
-  getSelectedService() {
-    const { services } = this.props;
-    const { selectedService } = this.state;
-    return selectedService || store.get('lastAtmSearchService') || services[0];
-  }
+  }, [
+    selectedSpanKind,
+    selectedTimeFrame,
+    getSelectedService,
+    fetchAllServiceMetrics,
+    fetchAggregatedServiceMetrics,
+  ]);
+
+  const handleServiceChange = useCallback((value: string) => {
```

Comment:
please keep functions in their original positions, it is difficult to follow the diff